jaxdecomp/_src/transpose.py

```diff
@@ -44,16 +44,21 @@ def abstract(x, kind, pdims, global_shape):
     assert kind in ['x_y', 'y_z', 'z_y', 'y_x']
     match kind:
     # From X to Y the axis are rolled by 1 and pdims are swapped wrt to the input pdims
```

Comment:
The process dims are being transposed at every step.
But I need the original pdims for my compiled primitive (the grid descriptor and the tranpose operator are indexed using `gdim` and `pdim` ... so even after a tranpose I need them back to find my primitive in `m_TransposeDescriptors32` or `m_TransposeDescriptors64`)